doc/lsst.pex.config/overview.rst

```diff
@@ -0,0 +1,140 @@
+The ``flatScalingType`` field is a `ChoiceField` type.
+In this case, a user can only set ``flatScalingType`` to a value of ``"USER"``, ``"MEAN"``, OR ``"MEDIAN"``.
+
+In code, a config object might be used like this:
+
+.. code-block:: python
+
+   def doIsrTask(ccd, configOverrideFilename=None):
+       config = IsrTaskConfig()
+       if configOverrideFilename is not None:
+           config.load(configOverrideFilename)
+           # Note: config override files are Python code with .py extensions
+           # by convention
+       config.validate()
+       config.freeze()
+
+       detectSaturation(ccd, config.fwhm, config.saturatedMaskName)
+       # Note: methods typically do not need the entire config; they should be
+       # passed only relevant parameters
+       for k in config.keysToRemoveFromAssembledCcd:
+           ccd.metadata.remove(k)
+       if config.doWrite:
+           ccd.write()
+
+Notice how configuration field values are accessible as attributes on the ``config`` instance.
+
+Also notice the `~Config.load` method.
+This is a way of loading configuration values from a file.
+A configuration override file for ``IsrTaskConfig`` might look like this:
+
+.. code-block:: python
+
+   root.doWrite = False
+   root.fwhm = 0.8
+   root.saturatedMaskName = 'SATUR'
+   root.flatScalingType = 'MEAN'
+   root.keysToRemoveFromAssembledCcd = ['AMPNAME']
+
+This override file looks like Python code because *it is*.
+The ``root`` variable refers to the config instance that called its `~Config.load` method, which is the ``config`` variable in the ``doIsrTask`` example.
+In more advanced cases a configuration field's value can itself be a config instance, so there will be a hierarchical namespace of configurations, like:
+
+.. code-block:: python
+
+   root.configField.fieldOnConfigField = 'value'
+
+Principles for using lsst.pex.config
+====================================
+
+:ref:`lsst.pex.config` arose from a desire to have a configuration object holding key-value pairs that also allows for (arbitrarily simple or complex) validation of configuration values.
+
+To configure code using :ref:`lsst.pex.config`, you create a subclass of the `Config` class.
+The subclass specifies the available `Field` attributes, their default values (if any), and their validation, if necessary.
+
+`Config` configuration objects are hierarchical (see `ConfigField`), so calling code can embed the configuration definitions of called code.
+
+Configurations are *not* input data.
+They should not be used in place of function or method arguments, nor are they intended to replace ordinary dictionary data structures.
+A good rule of thumb is that if a particular parameter does not have a useful default, it is probably an input rather than a
+configuration parameter.
+Another rule of thumb is that configuration parameters should generally not be set in algorithmic code, only in initialization or user interface code.
```

Comment:
I'd state this a bit more strongly: "In fact, changing configuration after a configurable object (such as `~lsst.pipe.base.Task`) has been initialized can lead to incorrect behavior."